source/adapters/cuda/usm.cpp

```diff
@@ -151,18 +151,22 @@ ur_result_t USMDeviceAllocImpl(void **ResultPtr, ur_context_handle_t,
 }
 
 ur_result_t USMSharedAllocImpl(void **ResultPtr, ur_context_handle_t Context,
-                               ur_device_handle_t Device,
+                               ur_device_handle_t CommandDevice,
                                ur_usm_host_mem_flags_t,
                                ur_usm_device_mem_flags_t, size_t Size,
                                uint32_t Alignment) {
   try {
-    ScopedContext Active(Device);
+    ScopedContext Active(CommandDevice);
     UR_CHECK_ERROR(cuMemAllocManaged((CUdeviceptr *)ResultPtr, Size,
                                      CU_MEM_ATTACH_GLOBAL));
-    for (const auto &device : Context->getDevices()) {
-      if (getAttribute(device, CU_DEVICE_ATTRIBUTE_CONCURRENT_MANAGED_ACCESS)) {
+    if (getAttribute(CommandDevice, CU_DEVICE_ATTRIBUTE_CONCURRENT_MANAGED_ACCESS)) {
+        UR_CHECK_ERROR(cuMemAdvise((CUdeviceptr)*ResultPtr, Size,
+                                   CU_MEM_ADVISE_SET_ACCESSED_BY, CommandDevice->get()));
+      }
+    for (const auto &Dev : Context->getDevices()) {
+      if (getAttribute(Dev, CU_DEVICE_ATTRIBUTE_CONCURRENT_MANAGED_ACCESS)) {
```

Comment:
It feelks like we should cache the result for this property on device initialisation to save us from querying it at this point regardless of how cheap it likely is. It is also used is some other USM managed memory entry points too.
That's a just a note, can totally be a separate PR.

Reply:
I'm not sure it needs to be used again? Where are you thinking it would be reused?

Reply:
Yeh pretty much my thinking and set in the `ur_device_handle_t_` constructor.

Reply:
But yeh probs to just merge this asap, so people can actually use malloc_shared in sycl cuda backend and the performance not being way worse than cuda runtime.